lib/internal/fs/promises.js

```diff
@@ -692,6 +702,10 @@ async function readFile(path, options) {
   if (path instanceof FileHandle)
     return readFileHandle(path, options);
 
+  if (options.signal?.aborted) {
+    throw lazyDOMException('The operation was aborted', 'AbortError');
+  }
+
   const fd = await open(path, flag, 0o666);
   return PromisePrototypeFinally(readFileHandle(fd, options), fd.close);
```

Comment:
What if the signal is aborted while we wait for `open`? Shouldn't we have the same `try { } finally { }` pattern as in `writeFile`?

Reply:
readFileHandle already throws

Reply:
So does `writeFileHandle`, maybe we should remove the change above (line 678..).

Reply:
While true, there's a minor difference here which why I changed it (as that was my first instinct as well). `writeFileHandle` returns without error if the amount of data to be write is `0` bits. I assumed that that's why the function was written in the original way (with the abort signal check before the write). If this change is immaterial, I think that removing the changes of line 678 is the best solution to this PR.

Reply:
Oh, re-reading it I assume that it was written this way to avoid allocating the buffer which might be large...

Reply:
Yeah, removing it is probably the right move here

Reply:
> Oh, re-reading it I assume that it was written this way to avoid allocating the buffer which might be large...

OK, so maybe we can add a comment explaining why it is done differently.

Reply:
I removed the signal check, and left the original `PromisePrototypeFinally`. From what I understand, the `Uint8Array` constructor shouldn't allocate much because it only offers a "view" into the buffer that it receives, so unless I'm misunderstanding something - it shouldn't really matter.

Reply:
@aduh95 I missed that `return PromisePrototypeFinally(readFileHandle(fd, options), fd.close);` didn't even pass the signal into the function. I've pushed a fix...